Core/src/org/sleuthkit/autopsy/contentviewers/MediaPlayerPanel.java

```diff
@@ -531,22 +549,236 @@ protected void done() {
 
         @Override
         public void actionPerformed(ActionEvent e) {
-            if (!progressSlider.getValueIsAdjusting()) {
-                long position = gstPlayBin.queryPosition(TimeUnit.NANOSECONDS);
-                long duration = gstPlayBin.queryDuration(TimeUnit.NANOSECONDS);
-                /**
-                 * Duration may not be known until there is video data in the
-                 * pipeline. We start this updater when data-flow has just been
-                 * initiated so buffering may still be in progress.
-                 */
-                if (duration >= 0 && position >= 0) {
-                    double relativePosition = (double) position / duration;
-                    progressSlider.setValue((int) (relativePosition * PROGRESS_SLIDER_SIZE));
+            try {
+                if (!progressSlider.getValueIsAdjusting()) {
+                    sliderLock.acquire();
+                    long position = gstPlayBin.queryPosition(TimeUnit.NANOSECONDS);
+                    long duration = gstPlayBin.queryDuration(TimeUnit.NANOSECONDS);
+                    /**
+                     * Duration may not be known until there is video data in
+                     * the pipeline. We start this updater when data-flow has
+                     * just been initiated so buffering may still be in
+                     * progress.
+                     */
+                    if (duration >= 0 && position >= 0) {
+                        double relativePosition = (double) position / duration;
+                        progressSlider.setValue((int) (relativePosition * PROGRESS_SLIDER_SIZE));
+                    }
+
+                    SwingUtilities.invokeLater(() -> {
+                        updateTimeLabel(position, duration);
+                    });
+                    sliderLock.release();
                 }
+            } catch (InterruptedException ex) {
+                //Video panel thread interrupted while waiting on lock.
```

Comment:
does the sliderLock get released if there's an exception?

Reply:
If this block is caught the lock was never acquired.